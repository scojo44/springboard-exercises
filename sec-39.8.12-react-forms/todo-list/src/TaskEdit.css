.TaskEdit {
  display: flex;
  align-items: center;
  width: 100%;
}
.TaskEdit input {
  width: 100%;
}