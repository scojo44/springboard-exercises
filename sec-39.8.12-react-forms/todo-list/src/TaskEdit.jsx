import React, { useState } from 'react'
import './TaskEdit.css'

function TaskEdit({id, task, update, completed}) {
  /** Send the updated task to the parent */
  function handleSubmit(e) {
    e.preventDefault();
    update(id, newTask, completed, false);
    setNewTask(() => '');
  }

  function handleChange(e) {
    setNewTask(() => e.target.value);
  }

  const [newTask, setNewTask] = useState(task);

  return (
    <form onSubmit={handleSubmit} className='TaskEdit'>
      <div><input type="text" name="edit" id="edit" value={newTask} onChange={handleChange} /></div>
      <button type="submit">Save</button>
    </form>
  )
}

export default TaskEdit
