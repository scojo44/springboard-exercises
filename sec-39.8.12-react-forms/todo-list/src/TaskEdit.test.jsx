import { render, fireEvent } from '@testing-library/react';
import TaskEdit from './TaskEdit';

describe('TaskEdit Tests', () => {
  it('renders without crashing', () => {
    render(<TaskEdit id={1} task="Learn Svelte" />);
  });

  it('matches snapshot', () => {
    const {asFragment} = render(<TaskEdit id={1} task="Learn Svelte" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call the passed update function when Save is clicked', () => {
    const update = vitest.fn();
    const {getByText, getByDisplayValue} = render(<TaskEdit id={1} task="Learn Svelte" update={update} completed={false} />);
    const input = getByDisplayValue('Learn Svelte');
    const button = getByText('Save');

    expect(input).toBeInTheDocument();
    expect(button).toBeInTheDocument();
    fireEvent.change(input, {target: {value: "Learn Preact"}})
    fireEvent.click(button);
    expect(update).toHaveBeenCalledWith(1, "Learn Preact", false, false);
  });
});
