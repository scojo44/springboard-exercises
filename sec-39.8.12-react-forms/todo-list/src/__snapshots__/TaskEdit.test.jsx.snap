// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TaskEdit Tests > matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="TaskEdit"
  >
    <div>
      <input
        id="edit"
        name="edit"
        type="text"
        value="Learn Svelte"
      />
    </div>
    <button
      type="submit"
    >
      Save
    </button>
  </form>
</DocumentFragment>
`;
